libs/api/bpf_syscall.cpp

```diff
@@ -6,188 +6,229 @@
 #include <windows.h>
 #include <WinError.h>
 #include <linux/bpf.h>
+#include <stdexcept>
 
-#define CHECK_SIZE(last_field_name)                                                    \
-    if (!tail_is_zero(                                                                 \
-            attr,                                                                      \
-            offsetof(union bpf_attr, last_field_name) + sizeof(attr->last_field_name), \
-            sizeof(union bpf_attr))) {                                                 \
-        return -EINVAL;                                                                \
-    }
-
-static bool
-tail_is_zero(const void* buf, unsigned int start, unsigned int end)
+template <typename T> class ExtensibleStruct
 {
-    const unsigned char* p = (const unsigned char*)buf + start;
-    const unsigned char* e = (const unsigned char*)buf + end;
+  private:
+    void* _orig;
+    size_t _orig_size;
+    T _tmp;
+    T* _p;
 
-    for (; p < e; p++) {
-        if ((*p) != 0) {
-            return false;
+    static void
+    check_tail(const void* buf, size_t start, size_t end)
```

Comment:
```suggestion
    check_tail(_In_reads_bytes_(end) const void* buf, size_t start, size_t end)
```

Reply:
Is there a doc somewhere which teaches me where and how to place these annotations?

Reply:
Found https://learn.microsoft.com/en-us/cpp/code-quality/annotating-function-parameters-and-return-values?view=msvc-170